symbolic-symcache/src/transform/mod.rs

```diff
@@ -2,30 +2,47 @@
 
 mod bcsymbolmap;
 pub use bcsymbolmap::*;
+use camino::{Utf8Path, Utf8PathBuf};
 
 #[cfg(feature = "il2cpp")]
 pub mod il2cpp;
 
 use std::borrow::Cow;
 
+use crate::normalize_path;
+
 /// A Function record to be written to the SymCache.
 #[non_exhaustive]
 pub struct Function<'s> {
     /// The functions name.
     pub name: Cow<'s, str>,
     /// The compilation directory of the function.
-    pub comp_dir: Option<Cow<'s, str>>,
+    pub comp_dir: Option<Cow<'s, Utf8Path>>,
```

Comment:
Can the public interface here stay strings, and we just use `Utf8Path` internally? That way, users of this crate, such as symbolicator, do not have to deal with a proprietary type.